zerver/actions/submessage.py

```diff
@@ -94,4 +95,9 @@ def do_add_submessage(
         UserMessage.objects.filter(message_id=message_id).values_list("user_profile_id", flat=True)
```

Comment:
Mentioning it here explicitly:

Didn't apply filter to these `target_user_ids` to check if they are still subscribed or not because as [mentioned in CZO](https://chat.zulip.org/#narrow/stream/3-backend/topic/Tighten.20access.20while.20sending.20events/near/1948787):
> if we see an easy way to send live-updates for messages that a user received while they were subscribed, and the user still has access to the message, it'd be nice to do that.

> I just wouldn't want to make these basic operations significantly slower to do an expensive access-control check involving a lot of extra database queries in non-protected history channels for this sort of live-update detail.

Since the usermessage row is deleted once a message is moved to a private stream which the user can't access, there isn't any bug that we'll send event for a message to a user which they can't access.

Reply:
Well the goal of #29380 is to add hardening such that if that invariant is violated by some bug, it has minimal user-impact. So I think it'd be preferable to define a common function for users who can get update events for a message that is careful about guests and private channels.

Reply:
Okay, Added a shared function `event_recipient_ids_for_action_on_messages`